src/adapter.rs

```diff
@@ -134,47 +155,94 @@ impl BluetoothAdapter for Bluez{
     fn get_address(&self) -> Result<String, Box<Error>> {
         self.0.get_address()
     }
+    fn set_address(&self, address: String) -> Result<(), Box<Error>>{
+        Err(Box::from(NOT_SUPPORTED_ERROR))
+    }
```

Comment:
These `set` implementations can become default implementation in the trait declarations, and then they don't need to be duplicated for every type: https://doc.rust-lang.org/book/ch10-02-traits.html#default-implementations